src/main/java/com/example/spring/service/cart/CartServiceImpl.java

```diff
@@ -97,7 +99,21 @@ public void createShoppingCart(User user) {
         shoppingCartRepository.save(shoppingCart);
     }
 
-    private ShoppingCart findByUserId(Long id) {
+    @Override
+    @Transactional
+    public void deleteCartItems(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "User with id" + userId + "not found!"));
+        List<CartItem> cartItems = new ArrayList<>(shoppingCart.getCartItems());

```

Comment:
You can find cartItem 
`CartItem cartItem = itemRepository.findByIdAndShoppingCartId(itemId, shoppingCart.getId())`

Reply:
Yeah, but i wanted to clear all cartItems after creating an order.
User has only 1 constant shopping cart, thats why i think its a good decision to do the way i did.

Reply:
You can just create method cart.clearCart(); in ShoppingCart 
```
public void clearCart() {
        cartItems.clear();
    }
```

And avoid quering DB